### DISCUSSION TITLE
How to keep images in pdf for a epub --> pdf conversion via ms engine?

### DISCUSSION THREAD

**Discussion by emk2203** [jgm/pandoc#10322] [Q&A] [answered]
I am trying to convert epub to pdf via simple `pandoc file.epub -t ms -o file.pdf` conversion. This turns the cover image into a placeholder: `[IMAGE: cover]`.

[I know that groff (ms) could deal with images](https://l04db4l4nc3r.github.io/groff-cheatsheet/examples/preprocessing/photos/PDFPIC/) via a `.PDFPIC` register: `.PDFPIC -[L|C|R] [Width[Height]] "image_path.pdf"` without the use of macros. just a simple

```
.PDFPIC -C "arch.pdf"
.PDFPIC -I 1i "arch.pdf" 2i 1i
```
is the example file given.

Is there a way to use this in pandoc workflow to keep image(s), at least the cover image, for a epub --> pdf conversion?

I am using pandoc 3.13 on Ubuntu 24.10.

**Comment by jgm** [comment, marked as answer]
If the image is already a PDF, `.PDFPIC` will be used.
You could use a Lua filter, perhaps, to convert non-PDFs in the mediabag to PDFs.

**Reply by emk2203** [reply]
This is a good strategy for external images.

What I am interested in is the transfer of internal, embedded images in the epub -- they would be in .jpeg or .png format according to spec. Is there a way to keep those? The tricky bit would be to keep track of their position in the document in case of extraction and subsequent conversion.

**Reply by jgm** [reply]
The mediabag contains the internal images, so yes, this strategy would work for them.

**Reply by emk2203** [reply]
Thanks, I give it a try. I need to learn more about pandoc and the specific terms used, I guess.
